pkg/kubelet/kubelet.go

```diff
@@ -1152,120 +1152,183 @@ func (kl *Kubelet) pullImageAndRunContainer(pod *api.BoundPod, container *api.Co
 	return containerID, nil
 }
 
-func (kl *Kubelet) syncPod(pod *api.BoundPod, containersInPod dockertools.DockerContainers) error {
+type podContainerChangesSpec struct {
+	StartInfraContainer bool
+	InfraContainerId    dockertools.DockerID
+	ContainersToStart   map[int]empty
```

Comment:
I feel kinda weird using the index. WDYT of having it be a pointer to a container instead?